docs/adrs/0002-handling-migration-from-sfdx-to-sf-cli-in-cumulusci.md

```diff
@@ -0,0 +1,75 @@
+---
+date: 2023-07-28
+status: Accepted
+author: "@jstvz"
+---
+
+# ADR 2: Handling Migration from SFDX to SF CLI in CumulusCI
+
+## Context and Problem Statement
+
+## Context
+
+Salesforce has announced the deprecation of SFDX (v7) and is now focusing on the newer SF CLI (v2). They've said that SF CLI (v2) is smart enough to understand all SFDX commands, as well as new SF commands. However, our web applications and other integrations depend on SFDX. SFDX is end-of-life, so we need to cut over and make some improvements by using the new, better features.
+
+## Decision
+
+### Considered Options
+
+1. **Run commands based on what the user has (SFDX or SF CLI):**
+
+    - Good: Gives users still using SFDX more time to switch.
+    - Good: Gives us time to fully understand the new SF CLI.
+    - Bad: It would make our code more complex and might confuse users.
+    - Bad: It divides our attention between old and new versions.
+
+2. **Stop supporting SFDX right away:**
+
+    - Good: Lets us focus on the new SF CLI and stop worrying about old code.
+    - Good: Speeds up our switch to the new CLI and helps us find and fix problems sooner.
+    - Bad: It could disrupt users' work if they're still using SFDX.
+    - Bad: There could be problems with the new SF CLI we don't know about yet.
+
+3. **Let SF CLI handle old SFDX commands:**
+    - Good: Reduces changes to our commands, making the switch smoother for users.
+    - Good: Lets us switch to the new CLI slowly, reducing risk of big issues.
+    - Bad: Delays the inevitable. If they drop deprecated commands without us noticing, it would break functionality for our users.
+    - Bad: Could delay us learning and using new features of the SF CLI.
+
+### Decision Outcome
+
+Here's how we plan to make this change:
+
+-   **Maintain Backwards Compatibility:** Our priority is to ensure that all existing processes continue to function as expected. We will not modify the existing SFDX commands in our project until SF CLI announces the version in which these commands will be deprecated.
+
+-   **Delegate maintaining backwards compatibility to the SF CLI:** We will continue using existing commands as written and rely on SF CLI to interpret these commands. We think this will make the change easier and less disruptive.
```

Comment:
I want to make sure I understand these two points together. Are you saying that we will _invoke_ the tool as `sf`, but we will pass SFDX-style commands (such as `force:org:display`) until they are explicitly deprecated?

Reply:
Good question. It's more accurate to say that we'd rely on the
`@salesforce/cli` package's `sfdx` alias. I'll update the text to
clarify that.